lib/activesupport/all/activesupport.rbi

```diff
@@ -888,6 +888,14 @@ class ActiveSupport::TimeWithZone
   # of `ActiveSupport.to_time_preserves_timezone`.
   sig { returns(Time) }
   def to_time; end
+
+  # Uses Date to provide precise Time calculations for years, months, and days according to the proleptic Gregorian calendar.
+  # The result is returned as a new `TimeWithZone` object.
+  # The options parameter takes a hash with any of these keys: :years, :months, :weeks, :days, :hours, :minutes, :seconds.
+  # If advancing by a value of variable length (i.e., years, weeks, months, days), move forward from `time`, otherwise move forward
+  # from utc, for accuracy when moving across DST boundaries.
+  sig { params(options: T::Hash[Symbol, Integer]).returns(ActiveSupport::TimeWithZone) }
```

Comment:
Does this only work with `Integer`, or do `Float` values work as well

Reply:
Looks like either works.